software/motion.py

```diff
@@ -0,0 +1,61 @@
+import math
+import numpy as np
+import tkinter as tk
+import serial
+import struct
+import serial.tools.list_ports
+import time as time
+
+class IRobotMotion:
+    def open(self):
+        pass
+    def close(self):
+        pass
+    def move(self, x_speed, y_speed, rot_speed):
+        pass
+
+class OmniMotionRobot(IRobotMotion):
+
+    def __init__(self):
+        pid = 22336
+        for x in serial.tools.list_ports.comports():
+            if x.pid == pid:
+                self.serialObj = serial.Serial(x.device)
```

Comment:
Just a heads up that it instantly opens the serial port so you should never create more than one instance of this class. The open method is there to avoid accidental serial port opening.